tests/APIv0/SmokeTest.php

```diff
@@ -468,40 +448,152 @@ public function testPostDiscussionFromDraft() {
+            'DeliveryType' => 'VIEW',
+            'DeliveryMethod' => 'JSON',
+            'Type' => 'Draft',
+            'LastCommentID' => 0,
+        ];
+
+        $r = $api->post("/post/comment/?discussionid={$discussionID}.json", $comment);
+        $responseCode = $r->getStatusCode();
+        $this->assertEquals(200, $responseCode);
+        $responseBody = $r->getBody();
 
         $draftModel = new \DraftModel();
-        $postedDraft = $draftModel->getWhere(['DraftID' => $responseBody['DraftID']])->firstRow(DATASET_TYPE_ARRAY);
+        $postedComment = $draftModel->getWhere(['DraftID' => $responseBody['DraftID']])->firstRow(DATASET_TYPE_ARRAY);
+        $this->assertEquals($postedComment['DiscussionID'], $comment['DiscussionID']);
+        $this->assertEquals($postedComment['Body'], $comment['Body']);
 
-        $r2 = $api->get("drafts/delete/{$postedDraft['DraftID']}/{$user['tk']}");
-        $statusCode2 = $r2->getStatusCode();
-        $this->assertEquals(200, $statusCode2);
+        return $postedComment;
+    }
+
+    /**
+     * Test posting a comment draft
+     *
+     * @depends testRegisterBasic
+     * @large
+     */
+    public function testPostCommentFromDraft() {
+        $api = $this->api();
+        $api->setUser($this->getTestUser());
+
+        $draft = $this->testSavingCommentDraft();
+
+        $postComment = [
+            'DiscussionID' => $draft['DiscussionID'],
+            'CommentID' => '',
+            'DraftID' => $draft['DraftID'],
+            'Format' => 'Markdown',
+            'Body' => 'Test comment draft',
+            'DeliveryType' => 'VIEW',
+            'DeliveryMethod' => 'JSON',
+            'Type' => 'Post',
+            'LastCommentID' => 0,
+        ];
+
+        $r1 = $api->post("/post/comment/?discussionid={$draft['DiscussionID']}.json", $postComment);
+        $responseCode1 = $r1->getStatusCode();
+        $this->assertEquals(200, $responseCode1);
+        $responseBody1 = $r1->getBody();
+
+        $commentID = $responseBody1['CommentID'];
+        $r2 = $api->post("/post/comment2.json?commentid={$commentID}&inserted=1");
+        $responseCode2 = $r2->getStatusCode();
+        $this->assertEquals(200, $responseCode2);
+
+        $commentModel = new \CommentModel();
+        $dbComment = $commentModel->getWhere(['CommentID' => $commentID ])->firstRow(DATASET_TYPE_ARRAY);
+        $this->assertEquals($dbComment['DiscussionID'], $postComment['DiscussionID']);
+        $this->assertEquals($dbComment['Body'], $postComment['Body']);
+    }
+
+    /**
+     * Test that a category with restricted permissions can be created.
+     *
+     * @return array $body
+     * @large
+     */
+    private function createCategory() {
+        $api = $this->api();
+        $admin = $api->querySystemUser(true);
+        $api->setUser($admin);
+
+        $r = $this->api()->post('/vanilla/settings/addcategory.json', [
+            'Name' => 'Test Category ',
+            'UrlCode' => 'test',
```

Comment:
If you call this function more than once in this test suite, I'm pretty sure it's going to fail those subsequent times, because the URL code is not going to be unique. Is there anything you can do here to ensure categories created with this function at least have a chance at unique names/URL codes?

Reply:
I added a name and url param to the method, this should any possible collisions.